python/src/iceberg/schema.py

```diff
@@ -301,11 +336,10 @@ def _(obj: StructType, visitor: SchemaVisitor[T]) -> T:
 @visit.register(ListType)
 def _(obj: ListType, visitor: SchemaVisitor[T]) -> T:
     """Visit a ListType with a concrete SchemaVisitor"""
+
     visitor.before_list_element(obj.element)
-    try:
-        result = visit(obj.element.type, visitor)
-    finally:
```

Comment:
Why remove the try/finally? It seems safer to have it, although I'll admit that it would be weird for an exception to be thrown during operation.

Reply:
This was very confusing when debugging, and I don't think we should swallow (at least not the broad Exception) those in silence. When visiting an element caused an error, it would actually crash on something else because the state was broken.